ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/datalayer/datalayer.js

```diff
@@ -87,39 +95,61 @@
                 }
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.FIELD, formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.FIELD, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('elementHelpShown', function(event){
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.HELP, event.detail.formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.HELP, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('elementErrorShown', function(event){
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.ERROR, event.detail.formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.ERROR, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('submitStart', function(event){
-                let eventInfo = {
-                    eventName : FormEvents.SUBMIT,
-                    fieldTitle,
-                    fieldType,
-                    panelTitle,
-                    formTitle
-                }
-                eventInfo.timeSpentOnForm = ((new Date()).getTime() - startTime) / 1000;
+                containerState[formContainerID].submitted = true;
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: eventInfo
+                    eventInfo: {
+                        type: FormEvents.SUBMIT,
+                        target: event.detail.fieldId,
+                        originalTarget: event.detail.fieldId,
+                        payload: {
+                            fieldTitle: containerState[formContainerID].fieldTitle,
+                            fieldType: containerState[formContainerID].fieldType,
+                            panelTitle: containerState[formContainerID].panelTitle,
+                            formTitle : event.detail.formTitle,
```

Comment:
why not `containerState[formContainerID]. formTitle`

Reply:
They contain the same value. 
For readability and consistency, I've changed the assignment as suggested.